daemon/cluster/cluster.go

```diff
@@ -1070,16 +1070,21 @@ func (c *Cluster) ServiceLogs(ctx context.Context, input string, config *backend
 		return c.errNoManager(state)
 	}
 
+	var selector swarmapi.LogSelector
 	service, err := getService(ctx, state.controlClient, input)
 	if err != nil {
-		c.mu.RUnlock()
-		return err
+		task, err := getTask(ctx, state.controlClient, input)
+		if err != nil {
+			c.mu.RUnlock()
+			return errors.Wrapf(err, "no such service or task: %q", input)
```

Comment:
As with `getService`, this may return a variety of errors. We should check which it returned to make sure the error produced here isn't misleading.

```go
if GetHTTPErrorStatusCode(err) == http.StatusNotFound {
        return errors.Errorf("no such service or task: %q", input)
} else {
        return err
}
```

We should avoid wrapping the error because it would give us output like `no such service or task: "taskid": task taskid not found`.